ocean_lib/ocean/test/test_util.py

```diff
@@ -121,12 +134,75 @@ def test_get_contracts_addresses():
         assert value.startswith("0x"), "It is not a token address."
 
 
-# FIXME: add tests for:
-# to_base_18
-# to_base
-# from_base_18
-# from_base
-# get_dtfactory_address
-# get_bfactory_address
-# get_ocean_token_address
-# init_components
+def test_to_base_18():
+    res = to_base_18(1.0)
+    assert res == 1000000000000000000, "Incorrect conversion to wei."
+
+
+def test_to_base():
+    res = to_base(1.0, 10)
+    assert res == 10000000000, "Incorrect conversion to wei."
+    res = to_base(1.0, 2)
+    assert res == 100, "Incorrect conversion to wei."
+    res = to_base(1.0, 18)
+    assert res == to_base_18(1.0), "Incorrect conversion to wei."
+
+
+def test_from_base_and_from_base_18():
+    res = from_base(10000000000, 10)
+    assert res == 1.0, "Incorrect conversion to ETH."
+    res = from_base(100, 2)
+    assert res == 1.0, "Incorrect conversion to ETH."
+    res = from_base(to_base_18(1.0), 18)
+    assert res == from_base_18(to_base_18(1.0)), "Incorrect conversion to ETH."
+
+
+def test_get_dtfactory_address():
+    config = Config(os.getenv(ENV_CONFIG_FILE))
+    ConfigProvider.set_config(config)
+    addresses = util.get_contracts_addresses("ganache", config)
+    assert addresses
+    assert isinstance(addresses, dict)
+    assert "DTFactory" in addresses
+
+    address = get_dtfactory_address()
+    assert address[:2] == "0x"
+    assert address == addresses["DTFactory"]
+
+
+def test_get_bfactory_address():
+    config = Config(os.getenv(ENV_CONFIG_FILE))
+    ConfigProvider.set_config(config)
+    addresses = util.get_contracts_addresses("ganache", config)
+    assert addresses
+    assert isinstance(addresses, dict)
+    assert "BFactory" in addresses
+
+    address = get_bfactory_address()
+    assert address[:2] == "0x"
+    assert address == addresses["BFactory"]
+
+
+def test_get_ocean_token_address():
+    config = Config(os.getenv(ENV_CONFIG_FILE))
+    ConfigProvider.set_config(config)
+    addresses = util.get_contracts_addresses("ganache", config)
+    assert addresses
+    assert isinstance(addresses, dict)
+    assert "Ocean" in addresses
+
+    address = get_ocean_token_address()
+    assert address[:2] == "0x"
+    assert address == addresses["Ocean"]
+
+
+def test_init_components():
+    config = Config(os.getenv(ENV_CONFIG_FILE))
+    ConfigProvider.set_config(config)
+    init_components()
+    assert ConfigProvider.get_config() == config
+    assert (
+        get_web3_connection_provider(config.network_url).__dict__["endpoint_uri"]
```

Comment:
let's try to remove __dict__ from actual assertions. We use dict as a simple debugging tool, but we should avoid it for assertions.